subsys/bluetooth/audio/cap_commander.c

```diff
@@ -399,14 +400,302 @@ int bt_cap_commander_broadcast_reception_start(
+			LOG_DBG("Too many subgroups %u/%u", stop_param->num_subgroups,
+				CONFIG_BT_BAP_BASS_MAX_SUBGROUPS);
+			return false;
+		}
+
+		for (size_t j = 0U; j < i; j++) {
+			const union bt_cap_set_member *other = &param->param[j].member;
+
+			if (other == member) {
+				LOG_DBG("param->members[%zu] (%p) is duplicated by "
+					"param->members[%zu] (%p)",
+					j, other, i, member);
+				return false;
+			}
+		}
+	}
+
+	return true;
+}
 
 int bt_cap_commander_broadcast_reception_stop(
 	const struct bt_cap_commander_broadcast_reception_stop_param *param)
 {
-	return -ENOSYS;
+	struct bt_bap_broadcast_assistant_mod_src_param mod_src_param = {0};
+	struct bt_cap_commander_proc_param *proc_param;
+	struct bt_cap_common_proc *active_proc;
+	struct bt_conn *conn;
+	int err;
+
+	if (bt_cap_common_proc_is_active()) {
+		LOG_DBG("A CAP procedure is already in progress");
+
+		return -EBUSY;
+	}
+
+	if (!valid_broadcast_reception_stop_param(param)) {
+		return -EINVAL;
+	}
+
+	bt_cap_common_start_proc(BT_CAP_COMMON_PROC_TYPE_BROADCAST_RECEPTION_STOP, param->count);
+
+	broadcast_assistant_cb.mod_src = cap_commander_broadcast_assistant_mod_src_cb;
+	broadcast_assistant_cb.rem_src = cap_commander_broadcast_assistant_rem_src_cb;
+	broadcast_assistant_cb.recv_state = cap_commander_broadcast_assistant_recv_state_cb;
+	if (!broadcast_assistant_cb_registered &&
+	    cap_commander_register_broadcast_assistant_cb() != 0) {
+		LOG_DBG("Failed to register broadcast assistant callbacks");
+
+		return -ENOEXEC;
+	}
+
+	active_proc = bt_cap_common_get_active_proc();
+
+	for (size_t i = 0U; i < param->count; i++) {
+		const struct bt_cap_commander_broadcast_reception_stop_member_param *member_param =
+			&param->param[i];
+		struct bt_cap_commander_proc_param *stored_param;
+		struct bt_conn *member_conn =
+			bt_cap_common_get_member_conn(param->type, &member_param->member);
+
+		if (member_conn == NULL) {
+			LOG_DBG("Invalid param->member[%zu]", i);
+
+			return -EINVAL;
+		}
+		/* Store the necessary parameters as we cannot assume that the supplied
+		 * parameters are kept valid
+		 */
+		stored_param = &active_proc->proc_param.commander[i];
+		stored_param->conn = member_conn;
+		stored_param->broadcast_reception_stop.src_id = member_param->src_id;
+		stored_param->broadcast_reception_stop.num_subgroups = member_param->num_subgroups;
+		for (size_t j = 0U; j < CONFIG_BT_BAP_BASS_MAX_SUBGROUPS; j++) {
+			stored_param->broadcast_reception_stop.subgroups[j].bis_sync = 0;
+			stored_param->broadcast_reception_stop.subgroups[j].metadata_len = 0;
+		}
+	}
+
+	proc_param = &active_proc->proc_param.commander[0];
+
+	conn = proc_param->conn;
+	copy_broadcast_reception_stop_param(&mod_src_param, &proc_param->broadcast_reception_stop);
+
+	active_proc->proc_initiated_cnt++;
+
+	err = bt_bap_broadcast_assistant_mod_src(conn, &mod_src_param);
+	if (err != 0) {
+		LOG_DBG("Failed to stop broadcast reception for conn %p: %d", (void *)conn, err);
+
+		return -ENOEXEC;
```

Comment:
Should we just always return `-ENOEXEC` here? Could we check for known/expected errors and send them to the application?

Reply:
we already check for invalid parameters, or if there is already a CAP procedure running

Reply:
My point is that if `bt_bap_broadcast_assistant_mod_src` returns e.g. `-EBUSY` or `ENOTCONN`, isn't it better to send those the application and document these as possible return values? 

For example if it returns `-EBUSY`, then the application can simply retry with a `k_work` or something.